### DISCUSSION TITLE
Collisions with moving children of MutableCompoundShape

### DISCUSSION THREAD

**Discussion by DocAce** [jrouwe/JoltPhysics#673] [Q&A] [answered]
When a dynamic body collides with a hand-animated compound child the resulting impulse doesn't appear to be as correct as it does if the animated object has a physics body of its own. I assume that this is because the simulation doesn't have a velocity for the animated child. There doesn't seem to any way of setting a child shape's velocity, right?
Would it be reasonable to support that (I understand that you probably don't want to increase the size of SubShape), or should I better give these animated objects their own physics bodies?

**Comment by jrouwe** [comment, marked as answer]
> There doesn't seem to any way of setting a child shape's velocity, right?

There is indeed no way to set a child shape's velocity. This is very fundamental to how the physics engine works and the only way to get this to work properly is to create separate bodies for child shapes that need their velocity propagated to the simulation. 

Another possible way, one that only works if the velocity of the child shape is perpendicular to the contact normal, is to return the velocity through ```ContactSettings::mRelativeLinearSurfaceVelocity``` in ```ContactListener::OnContactAdded``` / ```ContactListener::OnContactPersisted```.

**Reply by DocAce** [reply]
As I suspected, thanks
